### PR TITLE
[DAT-301] Payments - Fix payment date

### PR DESCRIPTION
**Pull request by nreal**
# Background
The payment date is the creat invoice date, now with this fix, the payment date is DateTime.Now because when executing this process the payment is created

Ticket [DAT-301](https://makingsense.atlassian.net/browse/DAT-301)

### CODE REVIEW

File: Doppler.Sap.Test/BillingRequestHandlerTest.cs [hunk 1]
```diff
@@ -108,20 +98,10 @@ public async Task BillingRequestHandler_ShouldBeCreateBillingInSap_WhenQueueHasO
                 sapConfigMock.Object,
                 Mock.Of<ILogger<BillingRequestHandler>>(),
                 sapServiceSettingsFactoryMock.Object,
-                httpClientFactoryMock.Object,
+                httpClientFactoryMock,
```
Comment (andresmoschini) [inline Doppler.Sap.Test/BillingRequestHandlerTest.cs:101]: I am not sure if it is well done or not, but in general, we tend to use the suffix `Mock` for mock created wrappers (the old class), and nothing for the doubles:

```suggestion
                httpClientFactory,
```

File: Doppler.Sap.Test/Utils/HttpHelperExtension.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public static class HttpHelperExtension
+    {
+        public static IHttpClientFactory GetHttpClientMock(string content, HttpStatusCode statusCode, Mock<HttpMessageHandler> httpHandlerMock = null)
+        {
+            var httpClientFactoryMock = new Mock<IHttpClientFactory>();
+            var httpMessageHandlerMock = httpHandlerMock ?? new Mock<HttpMessageHandler>();
+
+            httpMessageHandlerMock.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
```
Comment (andresmoschini) [inline Doppler.Sap.Test/Utils/HttpHelperExtension.cs:18]: Nice!

File: Doppler.Sap.Test/Utils/HttpHelperExtension.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+using Moq.Protected;
+
+namespace Doppler.Sap.Test.Utils
+{
+    public static class HttpHelperExtension
+    {
+        public static IHttpClientFactory GetHttpClientMock(string content, HttpStatusCode statusCode, Mock<HttpMessageHandler> httpHandlerMock = null)
+        {
+            var httpClientFactoryMock = new Mock<IHttpClientFactory>();
+            var httpMessageHandlerMock = httpHandlerMock ?? new Mock<HttpMessageHandler>();
```
Comment (andresmoschini) [inline Doppler.Sap.Test/Utils/HttpHelperExtension.cs:15]: If I understand it well, you are accepting a `Mock<HttpMessageHandler>` or creating a new one.

But, in both cases, you are manipulating it setting up `SendAsync` behavior. In my opinion, it is broken the [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

Do you really need both options?

Comment (nreal) [reply Doppler.Sap.Test/Utils/HttpHelperExtension.cs:15]: You are right, I don't need both options, only I need that the `Mock<HttpMessageHandler>` should be created in the test for `assert ` purposes, I'll change it thanks

**Pull request merged**